### DISCUSSION TITLE
[4.0] Why does the path filter not allow spaces in filenames?

### DISCUSSION THREAD

**Discussion by pjdevries** [joomla/joomla-cms#34279] [General]
I just found out that the path filter, i.e. `Joomla\Filter\InputFilter::cleanPath()`, does not allow spaces. I wonder why, since spaces are perfectly valid in filenames in both Linux and Windows.

**Comment by dgrammatiko** [comment]
> I wonder why, since spaces are perfectly valid in filenames in both Linux and Windows.

Spaces are fine at the OS level but because most of the files exposed to the web need to be linked the name is gonna be different due to the encoding.
![Screenshot 2021-05-30 at 12 00 36](https://user-images.githubusercontent.com/3889375/120099991-ae387200-c13e-11eb-825f-be1d15430f33.png)

You can replicate it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI

**Reply by pjdevries** [reply]
That I understand. But it's not the responsibility of the path filter. In my opinion, the path filter should simply be responsible for checking if a path is a valid path. Checking if an URL is a valid URL is a different responsibility. It's not up to tge filter to reject a path based on the assumption that it will be used in a URL.
